### PR TITLE
OEL-255: Create Showcase Search View

### PR DESCRIPTION
**Pull request by GilNovacomm**
### Description

[Insert description here]

### Change log

- Added:
- Changed:
- Deprecated:
- Removed:
- Fixed:
- Security:

### Commands

```sh
[Insert commands here]

```

### CODE REVIEW

File: behat.yml.dist [hunk 1]
```diff
@@ -18,4 +19,4 @@ default:
       drupal:
         drupal_root: "${drupal.root}"
   formatters:
-    progress: ~
+    progress: ~
```
Comment (cmarcmar) [inline behat.yml.dist]: Empty line at the end of the file needs to be added.

**Review by cmarcmar — DISMISSED** [review]

File: behat.yml.dist [hunk 2]
```diff
@@ -3,6 +3,7 @@ default:
     default:
       paths:
         - "%paths.base%/tests/features"
+        - "%paths.base%/modules/oe_showcase_search/tests/features"
```
Comment (drishu) [inline behat.yml.dist]: remove this

File: behat.yml.dist [hunk 3]
```diff
@@ -3,6 +3,7 @@ default:
...
       contexts:
         - Drupal\DrupalExtension\Context\MinkContext
         - Drupal\DrupalExtension\Context\DrupalContext
```
Comment (drishu) [inline behat.yml.dist]: Lets add some pages (and elements if needed) from the behat test here (see example https://github.com/openeuropa/oe_theme/blob/3.x/behat.yml.dist#L23 )

File: composer.json [hunk 4]
```diff
@@ -7,8 +7,15 @@
     "prefer-stable": true,
     "require": {
         "php": ">=7.3",
+        "drupal/better_exposed_filters": "^5.0@beta",
         "drupal/core": "^8.9 || ^9.1",
-        "drupal/search_api": "^1.20"
+        "drupal/search_api": "^1.20",
+        "npm-asset/jquery-ui-touch-punch": "^0.2.3",
```
Comment (drishu) [inline composer.json]: why do we need touch punch ?

File: modules/oe_showcase_search/tests/features/showcase_search_sorting.feature [hunk 5]
```diff
@@ -0,0 +1,37 @@
+@api
```
Comment (drishu) [inline modules/oe_showcase_search/tests/features/showcase_search_sorting.feature]: move this into the root, all modules will be enabled so no use in doing this here

File: modules/oe_showcase_search/tests/features/showcase_search_sorting.feature [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+Feature: Showcase search sorting.
+  As a site administrator
+  I need to be able to sort content on the showcase search.
+
+  Background:
+    Given I am logged in as a user with the "administer account settings, administer blocks" permission
+  
+  Scenario: Add sorting form block to search
+    When I visit "admin/structure/block/add/views_exposed_filter_block%3Ashowcase_search-showcase_search_page/bartik?region=content"
```
Comment (drishu) [inline modules/oe_showcase_search/tests/features/showcase_search_sorting.feature]: use page alias, see my comment in behat.yml.dist

File: modules/oe_showcase_search/tests/features/showcase_search_sorting.feature [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    And I press "Save block"
+    Then I should see "The block configuration has been saved."
+    
+  Scenario: Test sorting on Showcase Search by title.
+    When I visit "showcase-search"
+    Then I should see "Showcase Search"
+    # Test sorting by title.
+    And I select "A-Z" from "Sort by"
+    And I press "Apply"
+    Then I should see "Abico Diam Jugis"
```
Comment (drishu) [inline modules/oe_showcase_search/tests/features/showcase_search_sorting.feature]: add a couple more pages, same for every time you filter

File: modules/oe_showcase_search/tests/features/showcase_search_sorting.feature [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+    Then I should see "Voco"
+    # Test sorting by published date.
+    And I select "Published on Asc" from "Sort by"
+    And I press "Apply"
+    Then I should see "Hendrerit"
+    And I select "Published on Desc" from "Sort by"
+    And I press "Apply"
+    Then I should see "Imputo Neo Sagaciter"
+    # Delete display
+    Then I visit "admin/structure/block/manage/exposedformshowcase_searchpage_test/delete"
```
Comment (drishu) [inline modules/oe_showcase_search/tests/features/showcase_search_sorting.feature]: deleting this is not needed, rather check also the pager

File: modules/oe_showcase_search/tests/src/Functional/ShowcaseViewTest.php [hunk 9]
```diff
@@ -0,0 +1,71 @@
+<?php
```
Comment (drishu) [inline modules/oe_showcase_search/tests/src/Functional/ShowcaseViewTest.php]: delete this, the behat one is enough

File: runner.yml.dist [hunk 10]
```diff
@@ -21,6 +21,8 @@ drupal:
     # Don't aggregate or some pages will not be showed properly.
     - "./vendor/bin/drush -y config-set system.performance css.preprocess 0"
     - "./vendor/bin/drush -y config-set system.performance js.preprocess 0"
+    # Move file assets.
+    - "cp -R default_images/. build/sites/default/files/."
```
Comment (drishu) [inline runner.yml.dist]: lets check if we can set the path for these to be inside the files folder, so we dont have this folder in root

**Review by drishu — CHANGES_REQUESTED** [review]
I'm thinking its a good idea to separate content from functionality so lets move the content folder into oe_showcase_default_content (lets see if we can keep the ids in the info file, if not, move those also)

File: composer.json (hunk 4 above)
Comment (GilNovacomm) [reply composer.json]: Touch punch is a library used by Better Exposed filters, it works without it (I havent tested it on a mobile view):
image

File: modules/oe_showcase_search/tests/src/Functional/ShowcaseViewTest.php (hunk 9 above)
Comment (GilNovacomm) [reply modules/oe_showcase_search/tests/src/Functional/ShowcaseViewTest.php]: Test removed as per meeting with Francesco.

File: modules/oe_showcase_search/tests/features/showcase_search_sorting.feature (hunk 8 above)
Comment (GilNovacomm) [reply modules/oe_showcase_search/tests/features/showcase_search_sorting.feature]: Test removed as per meeting with Francesco.

File: modules/oe_showcase_search/tests/features/showcase_search_sorting.feature (hunk 7 above)
Comment (GilNovacomm) [reply modules/oe_showcase_search/tests/features/showcase_search_sorting.feature]: Test removed as per meeting with Francesco.

File: modules/oe_showcase_search/tests/features/showcase_search_sorting.feature (hunk 6 above)
Comment (GilNovacomm) [reply modules/oe_showcase_search/tests/features/showcase_search_sorting.feature]: Test removed as per meeting with Francesco.

File: modules/oe_showcase_search/tests/features/showcase_search_sorting.feature (hunk 5 above)
Comment (GilNovacomm) [reply modules/oe_showcase_search/tests/features/showcase_search_sorting.feature]: Test removed as per meeting with Francesco.

File: behat.yml.dist (hunk 3 above)
Comment (GilNovacomm) [reply behat.yml.dist]: Path removed

File: behat.yml.dist (hunk 2 above)
Comment (GilNovacomm) [reply behat.yml.dist]: Path removed.

File: composer.json (hunk 4 above)
Comment (drishu) [reply composer.json]: Lets remove BEF for now, and figure it out at theming

**Review by drishu — APPROVED** [review]

**Review by Maxfire — APPROVED** [review]
Approved :+1:

**Pull request merged**